Assets/Scripts/AchievementsService.cs

```diff
@@ -558,14 +583,27 @@ private async void GetAndCacheData(string uri)
                 await System.Threading.Tasks.Task.Yield();
             }
 
+            asyncOp.completed += operation =>
```

Comment:
Should it set the completed callback before waiting? Will this fire if the asyncOp isDone when you set the .completed?

Reply:
Does there need to be any error handling in the situation where the operation finishes in an error, and doesn't have the result available?

Reply:
Only meant to resolve the conversation that was crossed out, still consider first comment in this chain

Reply:
Resolved in the form of a comment that precedes this one. Consequently, marking conversation resolved.